sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,67 @@
+from sympy.combinatorics.fp_groups import FpGroup, FpSubgroup, subgroup_quotient
+from sympy import sieve
+
+class PcGroup(FpGroup):
+    '''
+    Class implementing Polycyclic groups.
+    To-do : Docstring completion.
+            Add error and exception handling
+    '''
+    def __init__(fp_group, pc_series=None, pcgs=None):
+        self.fp_group = fp_group
+        self.pc_series = pc_series
+        self.pcgs = pcgs
+        self.rel_orders = self.relative_orders()
+
+    def compute_pcgs(self):
+        pcgs = []
+        for i in range(0, len(self.pc_series) - 1):
+            quotient = subgroup_quotient(self.fp_grp.free_group, self.fp_grp,self.pc_series[i], self.pc_series[i+1])
+            pcgs.append(quotient.random())
+        self.pcgs = pcgs
+
+    def compute_pc_series(self):
+        self.pc_series = self.fp_group.compute_polycyclic_series()
+
+    def relative_orders(self):
+        rel_orders = []
+        for i in range(0, len(self.pc_series) - 1):
+            first_grp = self.pc_series[i]
+            sec_grp = self.pc_series[i+1]
+            if isinstance(self.pc_series[i], list):
+                first_grp = fp_group.subgroup(self.pc_series[i])
```

Comment:
I would expect `pc_series` to contain groups, not lists. And it does contain lists, then at least there should another attribute containing the relevant groups - they are computed during `compute_polycyclic_series` and it would be very inefficient to compute them again.

Reply:
Any thoughts about this?

Reply:
Maybe we could store it as a dictionary with values as the presentation of the subgroup on new generators. I mean, the keys would be the elements of the current `pc_gens` and values could be the `FpGroups` on a new set of generators.

Reply:
That could work. I think it will depend on what's going to be most convenient to use, and you might need to write a bit more code to work out the best thing.